### DISCUSSION TITLE
Introducing Archway v5.0.0

### DISCUSSION THREAD

**Discussion by emperorjm** [archway-network/discussions#27] [Announcements]
We are pleased to announce that we’ve tagged a new version of the Archway protocol which has some major changes and therefore we’ve jumped from `v4.0.2` to `v5.0.0`.
A myriad of updates has been made, the most notable being the shift to `Cosmos SDK` version `0.47.5` from `v0.45.16`. This transition also allows us to upgrade both `wasmd` (from v0.33 to v0.43) and `ibc-go` (from v4 to v7).

Before introducing these changes to our Testnet and Mainnet, we will deploy these updates to our Development chain to facilitate comprehensive testing. This step is crucial, especially for developers, ensuring that contracts and frontend apps are rigorously vetted to eliminate potential issues. We will make an announcement once this development environment is up and running.

Now, let's explore the implications for the various stakeholders in the Archway ecosystem.

## Smart Contract Developer

- The way to create gov props for Sudo operations has changed. It now follows `x/gov v1` types. The legacy gov props for `x/wasmd` have been removed from the codebase.
- [cosmwasm_1_4](https://github.com/CosmWasm/cosmwasm/blob/f11b456d46073d0324d9e39ed617f7fcfdbe1b06/docs/CAPABILITIES-BUILT-IN.md) capabilities are now supported. This enables the `DistributionQuery::DelegationRewards`, `DistributionQuery::DelegationTotalRewards` and `DistributionQuery::DelegatorValidators` queries.

## Frontend Developer

- If your frontend queries interact with the governance module, ensure the support for `x/gov v1` types. Some legacy proposal types are still supported so, good luck!
- Some REST stuff might break as well. The documentation is parse and therefore there is no exact certainty to the things that might break and therefore it’s best to use GRPC when possible.

## Validator & Node Operators

One important reminder is to back up the private keys of your nodes. Some migrations could affect these keys, so it's crucial to ensure you can reimport them if any issues arise.

- v0.19.0 IAVL introduces a new "**fast**" index. It allows for faster queries and iterations. The first time the chain is started after the upgrade, this index is created. The creation process might take time. While the index is being created, node operators can observe the following in the logs: "*Upgrading IAVL storage for faster queries + execution on the live state. This may take a while*". The message is printed for every module that has a non-transient store. As a result, it gives a good indication of the progress of the upgrade.
- You can now export selective modules and do not have to rely on exporting the whole chain. Look at `archwayd export -h` for more details.
- The SDK version includes some key store migration for the CLI. Make sure you backup your private keys!
- The gentx validation cli cmd is now available under `archwayd genesis validate-gentx`
- The cli argument `--broadcast-mode block` has been removed. RIP  You can use `-b sync` and then `archwayd q tx <txhash>` to get the response.
- Rosetta is now enabled at the protocol level

## Relayer Operator

- The SDK version includes some key store migration for the CLI. Make sure you backup your private keys!
- The cli argument `--broadcast-mode block` has been removed.

## The Community

- You can now cancel unbonding delegations.
- Governance prop creation process has changed. Look at `archwayd tx gov submit-proposal -h` for more details.
- New governance module parameter called `MinInitialDepositRatio` has been added which mandates the proposal creator to pay minimum `x%` of the deposit. This will help counter spam. (Currently, this feature is disabled. Can be enabled post upgrade if needed)
- Archway now has a native `x/nft` module
- Archway now supports [x/group module](https://tutorials.cosmos.network/tutorials/8-understand-sdk-modules/3-group.html)
- The way to modify Archway `x/rewards` parameters has changed. It now follows the `x/gov v1 `type.

**Comment by 7Two1** [comment]
`$ cat archwayd_v5.0.0_checksums.txt`

```
0cc6224d86eaf6890233eebe935079ef9d73aafa97d0233a1c4f49de10461ddf  archwayd_darwin_arm64
2929845ab109d94b6ac72e855b10bec89e2ff6e2ab19195ef5aa47a10f3db8f8  archwayd_darwin_amd64
834b36d6ecfd01b08b346c7ce3a6861f641055a2da6d02a3b26b5f50f53e37f1  archwayd_linux_amd64.zip
834b36d6ecfd01b08b346c7ce3a6861f641055a2da6d02a3b26b5f50f53e37f1  archwayd_v5.0.0_linux_amd64.zip
8e8e91c1f51d6cc2264d714df308a1a7b034e3330aec78aacd47d61865e7be88  archwayd_darwin_all.zip
8e8e91c1f51d6cc2264d714df308a1a7b034e3330aec78aacd47d61865e7be88  archwayd_v5.0.0_darwin_all.zip
95c1f221c74f5bbdfc3f7f94c6c07028c255ac361078c4203bde0d0557dba4d5  archwayd_darwin_arm64.zip
95c1f221c74f5bbdfc3f7f94c6c07028c255ac361078c4203bde0d0557dba4d5  archwayd_v5.0.0_darwin_arm64.zip
b4483d601f138a5ca2724d7516a15c5786d16003131d9920c42a7689cd81996b  archwayd_linux_arm64.zip
b4483d601f138a5ca2724d7516a15c5786d16003131d9920c42a7689cd81996b  archwayd_v5.0.0_linux_arm64.zip
cede35953ac21a5e8e16902922d0c265a79fbd1910abf368459767191079ba55  archwayd_darwin_amd64.zip
cede35953ac21a5e8e16902922d0c265a79fbd1910abf368459767191079ba55  archwayd_v5.0.0_darwin_amd64.zip
e4915bb46dfd8e589e8f2952feefda64a974a26a867ee6dedac6fc9b92b9b357  archwayd_linux_arm64
f16dd1573563d951bca676c5e71a38446bc14f7d7fe15f7c18fcbafea027f934  archwayd_darwin_all
feb78499c6c7195fb77931aa163ab8e71b4247958a698df09214b9f15b888ca1  archwayd_linux_amd64
```

`$ sha256sum ./archwayd_linux_amd64`

`93b8de71d02f530c25fcb4a34765faef7c6c6956ea12937794e943dc7229215c  ./archwayd_linux_amd64`

Hiya, chksum mismatch ?

**Reply by emperorjm** [reply]
We are looking into this @7Two1. Thanks for reporting.

**Reply by jlehtimaki** [reply]
Hello @7Two1 ,

From our testing the checksums all match, e.g.
```
╰ sha256sum archwayd_linux_amd64

feb78499c6c7195fb77931aa163ab8e71b4247958a698df09214b9f15b888ca1  archwayd_linux_amd64

╰ sha256sum archwayd_linux_arm64

e4915bb46dfd8e589e8f2952feefda64a974a26a867ee6dedac6fc9b92b9b357  archwayd_linux_arm64
```

Seems like you are comparing against `v4.0.2` binary

**Reply by 7Two1** [reply]
@jlehtimaki Yup, oops!

Thank you for the update
